### PR TITLE
Add MSSQL Support

### PR DESCRIPTION
**Pull request by jvtrigueros**
I'm attempting to add support for `mssql`, I'm not super familiar with what flags are needed for GraalVM support, but I gave it my best shot.

I needed to modify the compilation flags quite a bit to get this to compile natively. In the end, it does work, but I'm sure there are probably some issues with my approach.

### CODE REVIEW

File: script/compile [hunk 1]
```diff
@@ -59,13 +66,21 @@ args=( "-jar" "$JAR"
        "-H:Log=registerResource:"
        "-H:EnableURLProtocols=http,https"
        "--enable-all-security-services"
+       "--allow-incomplete-classpath"
```
Comment (jvtrigueros) [inline script/compile:69]: Without this flag, the compilation would fail with a recommendation for adding this flag.

File: script/compile [hunk 2]
```diff
@@ -59,13 +66,21 @@ args=( "-jar" "$JAR"
...
        "-H:+JNI"
        "--verbose"
        "--no-fallback"
        "--no-server"
        "--report-unsupported-elements-at-runtime"
-       "-H:IncludeResources=org/hsqldb/.*\.properties" "-H:IncludeResources=org/hsqldb/.*\.sql"
+       "-H:+AddAllCharsets"
```
Comment (jvtrigueros) [inline script/compile:75]: This was needed so to support mssql's [collation](https://docs.microsoft.com/en-us/sql/relational-databases/collations/collation-and-unicode-support?view=sql-server-ver15).

File: script/compile [hunk 3]
```diff
@@ -59,13 +66,21 @@ args=( "-jar" "$JAR"
...
+       "-H:IncludeResources=org/hsqldb/.*\.properties"
+       "-H:IncludeResources=org/hsqldb/.*\.sql"
+       "-H:IncludeResourceBundles=com.microsoft.sqlserver.jdbc.SQLServerResource"
        "--initialize-at-run-time=org.postgresql.sspi.SSPIClient"
+       "--initialize-at-run-time=com.microsoft.sqlserver.jdbc.SQLServerColumnEncryptionAzureKeyVaultProvider"
+       "--initialize-at-run-time=com.microsoft.sqlserver.jdbc.SQLServerFMTQuery"
+       "--initialize-at-run-time=com.microsoft.sqlserver.jdbc.SQLServerBouncyCastleLoader"
+       "--initialize-at-run-time=com.microsoft.sqlserver.jdbc.SQLServerMSAL4JUtils"
```
Comment (jvtrigueros) [inline script/compile:83]: These were added as recommendations from compilation errors.

**Comment by borkdude** [comment]
Thanks a lot @jvtrigueros !

**Comment by borkdude** [comment]
@jvtrigueros It seems the PR broke the builds

https://app.circleci.com/pipelines/github/babashka/babashka-sql-pods/130/workflows/a1d716b0-88e7-4226-9771-aa56aca5924c/jobs/533

Maybe make (some of the) mssql settings depend on the env variable.

**Comment by jvtrigueros** [comment]
Oh no! I'll submit another PR. I'll see if I can set up CircleCI on my fork to ensure that the build is still working.

**Comment by borkdude** [comment]
@jvtrigueros Normally PRs should be built also on this project, so maybe try that first. I don't know why your first PR didn't have this problem.

**Comment by borkdude** [comment]
@jvtrigueros Already fixed.

**Comment by jvtrigueros** [comment]
@borkdude you're the best! Would you still like me to submit a PR adding the extra stuff on the `.circleci` config files? I don't think the binary is being built at the moment.

**Comment by borkdude** [comment]
@jvtrigueros Yeah, please do that. I'm not sure if we are allowed to distribute binaries for MSSQL though in the pod-registry? What's the license on that thing?

**Comment by jvtrigueros** [comment]
I'll have to check, but I'll get back to you on that.

**Comment by jvtrigueros** [comment]
Ok, it seems like `microsoft/mssql-jdbc` is licensed under [MIT](https://github.com/microsoft/mssql-jdbc#license), so we should be good to redistribute! 🙌

**Comment by borkdude** [comment]
That's good news. So feel free to PR the CircleCI config and we can start distributing it via the pod registry.

**Pull request merged**